### PR TITLE
Fix resizing optimization for larger notebooks

### LINKED ISSUE #19611: Layout is broken after action "New View for Notebook"
## Description

The "New View for Notebook" results in new views that have an odd narrow layout.

Image

## Reproduce

1. Open a notebook and right click on its context menu
2. Click "New View for Notebook"
3. Expand the width of the new view
4. See that the code window is stuck in its initial width, sometimes with a very visible screen blanking

## Expected behavior

The code window should expand to the new width.

## Context

- Operating System and version: Linux Fedora 44
- Browser and version: Chrome 151.0.7922.173
- JupyterLab version: 4.6.3

Thank you for your continued work on Jupyter Lab and Jupyter Notebook.

### PR DESCRIPTION
**Pull request by krassowski**
Lumino's `LayoutItem` caches the geometry it writes. The resize freeze interfered with the cache, corrupting the layout of every node positioned through a `LayoutItem`. Widget nodes that are not positioned via layout were unharmed and are frozen without side effects.
- This PR fixes the issue by skipping the lumino-positioned widgets
- it reclaims the cost of the fix by explicitly freezing height on all elements; this means resizing gets even faster
- it moves some repetitive logic to helper functions and fixes minor edge cases with drag handling and timer disposal
- the bulk of the changeset are new tests

## References

- Fixes #19611.
- Follow-up to #18877, which added `OptimizedDockPanelSvg` and was released in 4.6.0.

## Code changes

- The freeze now skips elements Lumino positions, tested by their inline `position: absolute`, and applies to the plain DOM inside them. For a notebook that moves it from `.jp-Notebook` down to `.jp-WindowedPanel-outer`.
- Every frozen element now gets a definite `height`, where only the outermost one did before. A maximum alone still leaves the browser computing the height of a large text node from its content on every frame.
- The frozen sizes are re-measured as the drag moves. Two of those re-measures could run at once, only one of them could be cancelled, and the other wrote pixel sizes onto elements that had already been unfrozen.
- A second pointer going down during a drag no longer ends the drag early and leaves the panels frozen for good.
- The freeze now ends exactly when Lumino's drag ends: on a left-button `pointerup`, or on Escape. Any `pointerup` and any key press used to end it, leaving the rest of the drag unoptimized.
- `pointercancel` now ends the freeze too. Lumino does not listen for it but a pen or touch drag ends when a system gesture takes over.
- `dispose()` and `onAfterDetach()` now end the drag, so the re-measure timer stops instead of running on for the life of the page.

The benchmark from #18877 renders one very heavy cell. The freeze also has to handle an output area holding many modest outputs, so it now raises `maxNumberOutputs` and measures that shape too.

## User-facing changes

Previously, dragging a split handle in the main area could leave a notebook stuck at its pre-drag width, either narrower than its panel and centred in it, or wider and clipped. Expanding the browser window did not always recover it. This PR fixes it.

### Before

[dockpanel-resize-before.webm](https://github.com/user-attachments/assets/0dd012f1-2518-4918-8ffc-075b54846025)

### After

[dockpanel-resize-after.webm](https://github.com/user-attachments/assets/77110cf9-a879-4f1c-80e0-92504fcb3805)

To reproduce the issue one needs a notebook over the heaviness threshold (a thousand rendered nodes), with those nodes spread across many cells rather than concentrated in one.

Resize is now even faster, because every frozen element now gets a definite height rather than the outermost one alone:

```bash
BENCHMARK_NUMBER_SAMPLES=20 jlpm test:benchmark --grep "Notebook Resize"
```

| Variant | Mode | runs | n | avg | median | sd | min | max |
|---|---|---|---|---|---|---|---|---|
| before this PR | optimized | 2 | 40 | 937 ms | 902 ms | 140 ms | 744 ms | 1292 ms |
| before this PR | unoptimized | 2 | 40 | 1952 ms | 1931 ms | 119 ms | 1766 ms | 2242 ms |
| this PR | optimized | 2 | 40 | 865 ms | 831 ms | 103 ms | 756 ms | 1240 ms |
| this PR | unoptimized | 2 | 39 | 1959 ms | 1956 ms | 116 ms | 1758 ms | 2244 ms |

That is 2.14x faster before and 2.35x faster after, on the medians.

## Backwards-incompatible changes

None

## AI usage

- **Yes**: Some or all of the content of this PR was generated by AI.
- **Yes**: The human author has carefully reviewed this PR and run this code (keep this PR "draft" until the answer is YES)
- AI tools and models used: Opus 5 and Fable 5

### CODE REVIEW

File: packages/application/src/dockpanel.ts [hunk 1]
```diff
@@ -248,17 +331,74 @@ export class OptimizedDockPanelSvg extends DockPanelSvg {
...
+      case 'pointercancel':
+        // An unrelated pointer being cancelled, a palm resting on a touch
+        // screen for instance, must not end a drag it was never part of.
+        return (event as PointerEvent).pointerId === dragPointerId;
+      case 'keydown':
+        return (event as KeyboardEvent).keyCode === 27;
+      default:
+        return false;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (krassowski) [reply packages/application/src/dockpanel.ts:392]: This suggestion would break things; the comments were misleading - c70e53a clarifies things and adds a test.

File: packages/application/src/dockpanel.ts [hunk 2]
```diff
@@ -248,17 +331,79 @@ export class OptimizedDockPanelSvg extends DockPanelSvg {
...
+      case 'pointerup':
+        // Deliberately not matched against `dragPointerId`. Lumino releases
+        // on any button 0 release and then drops its document listeners, so
+        // matching here would leave the panels frozen with no drag running
+        // and no further event able to unfreeze them.
+        return (event as PointerEvent).button === 0;
+      case 'pointercancel':
+        // Matched, because Lumino has no cancel case to keep in step with. An
+        // unrelated pointer being cancelled, a palm resting on a touch screen
+        // for instance, must not end a drag it was never part of.
```
Comment (MUFFANUJ) [inline packages/application/src/dockpanel.ts:385]: This notes that Lumino has no cancel case, but that seems to leave the underlying Lumino dock-panel drag active on this path. `_endDrag()` clears JupyterLab’s frozen-element state, but it does not release Lumino’s document listeners/drag state. Should the `pointercancel` release be added in Lumino too, or otherwise tracked?

**Review by MUFFANUJ — APPROVED** [review]
LGTM!

File: packages/application/src/dockpanel.ts (hunk 2 above)
Comment (krassowski) [reply packages/application/src/dockpanel.ts:385]: I opened https://github.com/jupyterlab/lumino/issues/847.
